packages/@lwc/engine-core/src/framework/reporting.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { noop } from '@lwc/shared';
+import { VM } from './vm';
+
+export const enum ReportId {
+    CrossRootAriaInSyntheticShadow,
+}
+
+type ReportingDispatcher = (reportId: ReportId, tagName: string, vmIndex: number) => void;
```

Comment:
We could add a lot more details to this dispatcher, but I wanted to keep things simple. I'm not even sure exactly what we're going to want to report yet, so this API may change.